src/Symfony/Component/Workflow/MarkingStore/MethodMarkingStore.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\MarkingStore;
+
+use Symfony\Component\Workflow\Marking;
+
+/**
+ * MethodMarkingStore stores the marking with a subject's method.
+ *
+ * This store deals with a "single state" or "multiple state" Marking.
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class MethodMarkingStore implements MarkingStoreInterface
+{
+    private $property;
+    private $singleState;
+
+    /**
+     * @param string $property Used to determine methods to call
+     *                         The `getMarking` method will use `$subject->getProperty()`
+     *                         The `setMarking` method will use `$subject->setProperty(string|array $places, array $context = array())`
+     */
```

Comment:
missing @param bool for singleState

Reply:
No need. We don't duplicate PHP doc when the parameter has a type hint and is obvious

Reply:
ok i did not knew that